### PR TITLE
MF-790: Add swr to programs widget

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This PR:

1. Adds SWR data fetching hooks to the Programs widget.
2. Refactors the existing components to reuse these hooks.
3. Adds a mutate call in the Programs Form so that the related widgets automatically revalidate when a new program enrollment is successfully recorded via the form.
4. Refactors existing tests to work within the SWR paradigm.

## Issue

https://issues.openmrs.org/browse/MF-790

### CODE REVIEW

File: packages/esm-patient-programs-app/src/programs/programs-detailed-summary.component.tsx [hunk 1]
```diff
@@ -79,68 +70,53 @@ const ProgramsDetailedSummary: React.FC<ProgramsDetailedSummaryProps> = () => {
...
-                        </TableRow>
+      <div className={styles.widgetCard}>
+        <div className={styles.programsHeader}>
+          <h4 className={`${styles.productiveHeading03} ${styles.text02}`}>{headerTitle}</h4>
+          <span>{isValidating ? <InlineLoading /> : null}</span>
+          <Button kind="ghost" renderIcon={Add16} iconDescription="Add programs" onClick={launchProgramsForm}>
+            {t('add', 'Add')}
+          </Button>
+        </div>
+        <TableContainer>
```
Comment (HerbertYiga) [inline packages/esm-patient-programs-app/src/programs/programs-detailed-summary.component.tsx:85]: cc @FlorianRappl @brandones

File: packages/esm-patient-programs-app/src/programs/programs-form.component.tsx [hunk 2]
```diff
@@ -89,87 +26,40 @@ interface ProgramsFormProps {
...
   }
 
   function handleProgramChange(event) {
-    dispatch({
-      type: ActionTypes.selectProgram,
-      program: event.target.value,
-      availablePrograms: viewState.availablePrograms,
-      eligiblePrograms: viewState.eligiblePrograms,
-    });
+    setSelectedProgram(event.target.value);
```
Comment (brandones) [inline packages/esm-patient-programs-app/src/programs/programs-form.component.tsx:49]: I think you can get rid of the one line function and just inline it

**Review by brandones — APPROVED** [review]
One piece of important feedback; feel free to merge after addressing that (assuming you agree).

**Comment by brandones** [comment]
Oh, and fix the failing check, obviously

**Pull request merged**